CONTRIBUTING.md

```diff
@@ -521,16 +521,23 @@ can be used in these cases.
 ### Feature Flags
 
 Sometimes we want to introduce new breaking behavior because we believe this is
-the correct default behavior for the CDK. The problem of course is that breaking
-changes are only allowed in major versions and those are rare.
+the correct default behavior for the CDK. However, breaking changes are only allowed 
+in major versions and those are rare.
 
 To address this need, we have a feature flags pattern/mechanism. It allows us to
 introduce new breaking behavior which is disabled by default (so existing
 projects will not be affected) but enabled automatically for new projects
 created through `cdk init`.
 
+**DISCLAIMER**: feature flags are a "last resort" mechanism and should only be used
+when it is impossible to figure out a way to introduce a feature without breaking
+existing users. If you think your feature should be implemented behind a feature
+flag, you will need to get an approval from at least two core team members. We do that
+in order to avoid the abuse of this powerful capability.
+
 The pattern is simple:
 
+1. Seek the approval of a core team member that a feature flag can be used.
```

Comment:
Feels like we should elaborate one how to seek core team approval with a couple of sub items on this one?

If the feature in question is being planned via an RFC, and the feature flag is contained in the proposal, core team member approval should include the feature flag.

If the feature is being tracked in a single issue without an RFC, approval can probably be granted there? If there is no existing issue for the feature, we already encourage contributors to create one in the contributing guide so that case is probably already accounted for.

Reply:
Added